src/composables/useGetParticipants.ts

```diff
@@ -0,0 +1,264 @@
+	const isDocumentVisible = useDocumentVisibility()
+
+	const isActive = computed(() => activeTab.value === 'participants')
+	const token = useGetToken()
+	const conversation = computed<Conversation | undefined>(() => store.getters.conversation(token.value))
+	const isInLobby = computed<boolean>(() => store.getters.isInLobby)
+	const isModeratorOrUser = computed<boolean>(() => store.getters.isModeratorOrUser)
+	const isOneToOneConversation = computed(() => conversation.value?.type === CONVERSATION.TYPE.ONE_TO_ONE
+		|| conversation.value?.type === CONVERSATION.TYPE.ONE_TO_ONE_FORMER)
+
+	/**
+	 * Initialise the get participants listeners
+	 *
+	 */
+	function initialiseGetParticipants() {
+		EventBus.on('joined-conversation', onJoinedConversation)
+		if (experimentalUpdateParticipants) {
+			EventBus.on('signaling-users-in-room', handleUsersUpdated)
+			EventBus.on('signaling-users-joined', handleUsersUpdated)
+			EventBus.on('signaling-users-changed', handleUsersUpdated)
+			EventBus.on('signaling-users-left', handleUsersLeft)
+			EventBus.on('signaling-all-users-changed-in-call-to-disconnected', handleUsersDisconnected)
+			EventBus.on('signaling-participant-list-updated', throttleUpdateParticipants)
+		} else {
+			// FIXME this works only temporary until signaling is fixed to be only on the calls
+			// Then we have to search for another solution. Maybe the room list which we update
+			// periodically gets a hash of all online sessions?
+			EventBus.on('signaling-participant-list-changed', throttleUpdateParticipants)
+		}
+		EventBus.on('signaling-users-changed', checkCurrentUserPermissions)
+	}
+
+	/**
+	 * Patch participants list from signaling messages (joined/changed)
+	 */
+	function handleUsersUpdated([users]: [SignalingSessionPayload[]]) {
+		if (sessionStore.updateSessions(token.value, users)) {
+			throttleUpdateParticipants()
+		} else {
+			throttleLongUpdate()
+		}
+	}
+
+	/**
+	 * Check if current user permission has been changed from signaling messages
+	 */
+	async function checkCurrentUserPermissions([users]: [StandaloneSignalingUpdateSession[]]) {
+		// TODO: move logic to sessionStore once experimental flag is dropped
+		const currentUser = users.find((user) => user.userId === actorStore.userId)
+		if (!currentUser) {
+			return
+		}
+
+		if (currentUser.participantPermissions !== conversation.value?.permissions) {
+			await store.dispatch('fetchConversation', { token: token.value })
+		}
+
+		const currentParticipant = store.getters.getParticipant(token.value, actorStore.attendeeId) as Participant | null
+		if (!currentParticipant) {
+			return
+		}
+
+		if (currentUser.participantPermissions !== currentParticipant.permissions) {
+			await cancelableGetParticipants()
+		}
+	}
+
+	/**
+	 * Patch participants list from signaling messages (left)
+	 */
+	function handleUsersLeft([sessionIds]: [StandaloneSignalingLeaveSession[]]) {
+		sessionStore.updateSessionsLeft(token.value, sessionIds)
+		throttleLongUpdate()
+	}
```

Comment:
This can be received for own leave or others
```suggestion
	async function handleUsersLeft([sessionIds]: [StandaloneSignalingLeaveSession[]]) {
		const isOwnSessionLeft = actorStore.sessionId && sessionIds.includes(actorStore.sessionId)
		sessionStore.updateSessionsLeft(token.value, sessionIds)
		if (isInCall.value && isOwnSessionLeft) {
			await cancelableGetParticipants()
		} else {
			throttleLongUpdate()
		}
	}
```

Reply:
Users left are for closing session with current room, when you navigate away. 
Why do you need to fetch participants list, if you're in the call and you're leaving this conversation? 👀